File: java/data_src_dependent/bst_games/bst_games.java

//
// SimpleDriver.java : Driver to illustrate importing USGS earthquake data
// 
import java.util.Scanner;
import bridges.connect.Bridges;
import bridges.base.BSTElement;
import bridges.data_src_dependent.Game;
import java.util.ArrayList;

public class bst_games {
	public static final int maxElements = 100; //number of tweets

	public static void main(String[] args) throws Exception{
	
    						// Instantiate a Bridges object 
		Bridges bridges = new Bridges (21, args[0], args[1]);
		bridges.setServer (args[2]);
		bridges.setTitle("Binary Search Tree with IGN Game Data");

							// Retrieve Game Data
		ArrayList<Game> game_list = 
				(ArrayList<Game>) Bridges.getGameData();

		BST<String, Game>  bst = new BST<String, Game>();
//		for ( int i = 0; i < game_list.size(); i++ ){
		for ( int i = 0; i < 5000; i++ ){
			bst.insert(game_list.get(i).getTitle(), game_list.get(i));
		}

							// give BRIDGES handle to tree root
		bridges.setDataStructure(bst.getTreeRoot());
							// visualize the tree
		bridges.visualize();
	}
}

File: python/data_src_dependent/bst_games/bst_games.py
from bridges.bridges import *
from bridges.data_src_dependent.data_source import *
from data_src_dependent.bst_games.BST import *
import sys

def main():
    args = sys.argv[1:]

    # create the Bridges object, set credentials

    bridges = Bridges(int(args[0]), args[1], args[2])

    if len(args) > 3:
        bridges.connector.set_server(args[3])

    bridges.set_title("Binary Search Tree with IGN Game Data")

    game_list = get_game_data()

    bst = BST()
    for i in range(5000):
        bst.insert(game_list[i].get_title(), game_list[i])

    bridges.set_data_structure(bst.get_tree_root())
    bridges.visualize()

if __name__ == '__main__':
    main()